pip/basecommand.py

```diff
@@ -266,6 +265,8 @@ def populate_requirement_set(requirement_set, args, options, finder,
         """
         Marshal cmd line args into a requirement set.
         """
+        # NOTE: As a side-effect, options.require_hashes may be updated
```

Comment:
I like that you've mentioned the possibly side effects here, calling out non-obvious properties of the current code is a great use of comments.